premium/usage.go

```diff
@@ -0,0 +1,264 @@
+package premium
+
+import (
+	"context"
+	"fmt"
+	cqapi "github.com/cloudquery/cloudquery-api-go"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"net/http"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	defaultBatchLimit            = 1000
+	defaultMaxRetries            = 5
+	defaultMaxWaitTime           = 60 * time.Second
+	defaultMinimumUpdateDuration = 10 * time.Second
```

Comment:
```suggestion
	defaultMinTimeBetweenFlushes = 10 * time.Second
```

(nit)

And we'd update the `With...` functions accordingly